packages/trpc/server/middlewares/captureErrorsMiddleware.ts

```diff
@@ -9,6 +11,7 @@ const captureErrorsMiddleware = middleware(async ({ next }) => {
     if (!cause) {
       return result;
     }
+    Sentry.captureException(cause);
```

Comment:
If the NEXT_PUBLIC_SENTRY_DSN is not set, this is a no-op. This is the same strategy the API project currently has so I decided to follow suit.

I considered adding a logger factory that returns a logger specific to Sentry when NEXT_PUBLIC_SENTRY_DSN is set, but decided against it because it introduces more classes and layering that seems necessary since this operation is harmless.

Happy to reconsider if someone feels strongly about it.